packages/lesswrong/components/ea-forum/giving-portal/GivingSeasonHeader.tsx

```diff
@@ -278,7 +429,76 @@ const GivingSeasonHeader = ({
   const isVotingPortal = pathname.startsWith("/voting-portal");
   // Show voting steps if we are on a path like /voting-portal/compare (with anything after /voting-portal/)
   const showVotingSteps = isVotingPortal && /\/voting-portal\/\w/.test(pathname);
-  const solidHeader = currentRoute?.path !== "/";
+  const showHearts = currentRoute?.path === "/";
+
+  const {data} = useQuery(heartsQuery, {
+    variables: {
+      electionName: eaGivingSeason23ElectionName,
+    },
+    skip: !showHearts,
+  });
+  const [hearts, setHearts] = useState<GivingSeasonHeart[]>(data?.GivingSeasonHearts ?? []);
+
+  const [addHeart, {loading: isAddingHeart}] = useMutation(heartsMutation, {
+    errorPolicy: "all",
+  });
+
+  const headerRef = useRef<HTMLDivElement>(null);
+
+  const normalizeCoords = useCallback((clientX: number, clientY: number) => {
+    if (headerRef.current) {
+      const bounds = headerRef.current.getBoundingClientRect();
+      if (
+        clientX > bounds.left &&
+        clientX < bounds.right &&
+        clientY > bounds.top &&
+        clientY < bounds.bottom
+      ) {
+        return {
+          x: clientX / bounds.width,
+          y: clientY / bounds.height,
+        };
+      }
+    }
+    return null;
+  }, [headerRef]);
+
+  const canAddHeart = !!currentUser && !!electionVote?.submittedAt && !isAddingHeart;
```

Comment:
Do we really want to limit this to only those who voted in the election? I feel like we should allow anyone to add a heart